packages/printoptions/init.lua

```diff
@@ -52,6 +66,21 @@ local function imageResolutionConverter (filename, widthInPx, resolution, pageno
   local sourceFilename = filename
   local basename, ext = handlePath(filename)
   local flatten = SILE.settings:get("printoptions.image.flatten")
+  local grayscale = SILE.settings:get("printoptions.image.grayscale")
+
+  local width, _, xres = SILE.outputter:getImageSize(sourceFilename, pageno or 1)
+  local actualWidthInPx = width * xres / 72
+  if actualWidthInPx <= widthInPx then
+    -- No need to resample, the image is already smaller than the target width.
```

Comment:
Still, should we grayscale under-resolved images instead of just bypassing any processing? Dunno!